torch/quantization/fx/_equalize.py

```diff
@@ -224,3 +218,259 @@ def node_supports_equalization(node: Node, modules) -> bool:
+    if maybe_obs_node is None:
+        return None
+
+    maybe_eq_obs_node = maybe_get_next_module(maybe_obs_node, modules, _InputEqualizationObserver)
+    if maybe_eq_obs_node is None:
+        return None
+
+    maybe_eq_obs = modules[str(maybe_eq_obs_node)]
+    assert(isinstance(maybe_eq_obs, _InputEqualizationObserver))
+    return maybe_eq_obs
+
+def maybe_get_next_equalization_scale(node: Node, modules: Dict[str, nn.Module]) -> Optional[torch.Tensor]:
+    """ If the next next node is an InputEqualizationObserver then we want to
+    return its equalization scale, else we return 1
+
+    This is used in the case where there are two connecting linear layers:
+        linear1 -> LinearOutObs -> InputEqObs -> linear2
+    In this case, the node given is linear1 and we want to locate the InputEqObs.
+    """
+    next_inp_eq_obs = maybe_get_next_input_eq_obs(node, modules)
+    if isinstance(next_inp_eq_obs, _InputEqualizationObserver):
+        return next_inp_eq_obs.equalization_scale
+    return None
+
+def scale_input_observer(node: Node, modules: Dict[str, nn.Module]) -> None:
+    """ Scales the following input quantization observer's min/max values by
+    updating the values with the scaled min/max values calculated by the input
+    equalization observer
+    """
+    input_eq_obs = modules[str(node.target)]
+    assert(isinstance(input_eq_obs, _InputEqualizationObserver))
+
+    input_quant_obs_node = node.args[0]
+    assert(isinstance(input_quant_obs_node, Node))
+
+    input_quant_obs = modules[str(input_quant_obs_node.target)]
+    if not isinstance(input_quant_obs, ObserverBase):
+        return
+
+    min_input_scaled, max_input_scaled = input_eq_obs.calculate_scaled_minmax()
+    input_quant_obs.min_val = min_input_scaled
+    input_quant_obs.max_val = max_input_scaled
+
+def scale_weight_node(
+    node: Node,
+    modules: Dict[str, nn.Module],
+    equalization_scale: torch.Tensor,
+    next_equalization_scale: Optional[torch.Tensor],
+) -> None:
+    """ Scale the weights for input-weight equalization by multiplying the
+    weight by 1/equalization_scale and next_equalization_scale
+
+    Args:
+        node: Current node whose weights we want to scale
+        equalization_scale: Current node's calculated equalization scale
+        next_equalization_scale: Next node's calculated equalization scale if
+           the following node needs to be equalized, 1 otherwise
+    """
+    assert(isinstance(node.target, str))
+
+    # Scale the weights for input-weight equalization
+    # If the following layer needs to be equalized then we will multiply its scale
+    weight = modules[node.target].weight
+    assert(isinstance(weight, torch.Tensor))
+
+    scaled_weight = torch.mul(weight, torch.reciprocal(equalization_scale))
+
+    if next_equalization_scale is None:
+        modules[node.target].weight = nn.Parameter(scaled_weight)
+        return
+
+    scaled_weight = torch.mul(scaled_weight, next_equalization_scale)
```

Comment:
Is there any way we can combine the mul on line 343 and 349 into the same `torch.mul` call?

Reply:
we could replace line 343 with:
```python
scaled_weight = torch.mul(weight, torch.reciprocal(equalization_scale)) if next_equalization_scale is None else torch.mul(torch.mul(weight, torch.reciprocal(equalization_scale)), next_equalization_scale)
```

or maybe something like....
```python
next_eq_scale = 1 if next_equalization_scale is None else next_equalization_scale
scaled_weight = torch.mul(torch.mul(weight, torch.reciprocal(equalization_scale)), next_eq_scale)
```

I think I originally had it that `maybe_get_next_equalization_scale` returned 1 if it didn't exist, but we changed it to return a None instead.

Reply:
Yeah, I think this is fine. I was thinking if we could combine `equalization_scale` and `next_equalization_scale` (if it exists) into a single scale value and use a single `mul` op for it. Maybe that isn't as straightforward.